### PR TITLE
feat(external_api) add command and event listener for CS

### PR DESCRIPTION
**Pull request by horymury**
@quitrk @hristoterezov @saghul 

Documentation update PR: https://github.com/jitsi/handbook/pull/135/files

### CODE REVIEW

File: react/features/base/tracks/middleware.js [hunk 1]
```diff
@@ -64,7 +72,19 @@ MiddlewareRegistry.register(store => next => action => {
         return result;
     }
     case TRACK_REMOVED: {
-        _removeNoDataFromSourceNotification(store, action.track);
+        const { track } = action;
+        const { jitsiTrack } = track;
+
+        if (jitsiTrack.videoType === 'desktop' && typeof APP === 'object') {
+            const participantId = _getJitsiTrackParticipantId(store, jitsiTrack);
```
Comment (hristoterezov) [inline react/features/base/tracks/middleware.js:79]: Maybe you can use `getTrackByJitsiTrack` to get the track from the store. This way you will have the videoType and the participantID from the track object in the store. WDYT?

File: react/features/base/tracks/middleware.js [hunk 2]
```diff
@@ -176,6 +197,63 @@ MiddlewareRegistry.register(store => next => action => {
...
+/**
+ * Gets the participant ID of the given jitsi track.
+ *
+ * @param {Store} store - The redux store.
+ * @param {JitsiTrack} jitsiTrack - The jitsi media track.
+ * @private
+ * @returns {string} - The participant ID.
+ */
+function _getJitsiTrackParticipantId(store, jitsiTrack) {
+    return jitsiTrack.isLocal()
```
Comment (hristoterezov) [inline react/features/base/tracks/middleware.js:209]: I'm not sure you need this function at all but if you do maybe you can use `getTrackByJitsiTrack` and from there you can get the participantID. WDYT?

**Comment by hristoterezov** [comment]
Discussed with @horymury offline that he will try another approach that will be cleaner and simpler.

**Comment by horymury** [comment]
@hristoterezov refactored as discussed with using a subscriber on the store for detecting the currently sharing participants.

File: react/features/base/tracks/subscriber.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    /* selector */ state => state['features/base/tracks'],
+    /* listener */ tracks => {
+        if (typeof APP !== 'object') {
+            return;
+        }
+
+        const sharingParticipants = tracks.filter(tr => tr.videoType === 'desktop').map(t => t.participantId);
+
+        if (!_.isEqual(_.sortBy(currentlySharingParticipants), _.sortBy(sharingParticipants))) {
+            currentlySharingParticipants = sharingParticipants;
```
Comment (saghul) [inline react/features/base/tracks/subscriber.js:27]: Please don't store tracks there, the listener function also returns the previous selector value, you can use that instead, right?

**Review by saghul — CHANGES_REQUESTED** [review]

File: react/features/base/tracks/subscriber.js (hunk 3 above)
Comment (horymury) [reply react/features/base/tracks/subscriber.js:27]: well yes but then we need query for CS tracks on both previous and current tracks, which sound a bit more costly?

Comment (saghul) [reply react/features/base/tracks/subscriber.js:27]: Only if they change, so not that costly I think.

Comment (horymury) [reply react/features/base/tracks/subscriber.js]: refactored as discussed

**Review by saghul — APPROVED** [review]

**Pull request merged**